src/analysis/scoping_analysis.cpp

```diff
@@ -130,8 +133,18 @@ struct ScopingAnalysis::ScopeNameUsage {
     StrSet got_from_closure;
     StrSet passthrough_accesses; // what names a child scope accesses a name from a parent scope
 
+    // `import *` and `exec` both force the scope to use the NAME lookup
+    bool hasNameForcingSyntax;
+    AST_ImportFrom* nameForcingNodeImportStar;
+    AST_Exec* nameForcingNodeBareExec;
+
+    // If it has a free variable / if any child has a free variable
+    bool free;
+    bool child_free;
```

Comment:
`free` maybe. `child_free` is actually different, because this triggers, for example, because `h` has a free variable:

```python
def f():
    exec "a = 5"
    def g():
        b = 2
        def h():
            print b
```

I'm not sure if there's any reason for this to be invalid, but it is.